src/main/java/project/bokapp/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,51 @@
+package project.bokapp.repository;
+
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import project.bokapp.model.Book;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```

Comment:
as you're using lombok it's more logical to replace constructors with RequiredArgsConstructor annotation